storey/sources.py

```diff
@@ -1012,3 +1012,99 @@ def _init(self):
             else:
                 df = pandas.read_parquet(path, columns=self._columns, storage_options=self._storage_options)
             self._dfs.append(df)
+
+
+class SQLSource(_IterableSource, WithUUID):
+    """Use SQL table as input source for a flow.
+
+
+    :parameter key_field: the primary key of the table.
+    :parameter time_field: column to be used as time for events.
+    :parameter id_field: column to be used as ID for events.
+    :parameter db_name: url string connection to sql database.
+    :parameter table_name: the name of the table to access,
+                                from the current database
+
+    """
+
+    def __init__(
+        self,
+        db_path: str = None,
+        table_name: str = None,
+        key_field: Optional[Union[str, List[str]]] = None,
+        time_field: Optional[str] = None,
+        id_field: Optional[str] = None,
+        **kwargs,
+    ):
+
+        if key_field is not None:
+            kwargs["key_field"] = key_field
+        if time_field is not None:
+            kwargs["time_field"] = time_field
+        if id_field is not None:
+            kwargs["id_field"] = id_field
+        _IterableSource.__init__(self, **kwargs)
+        WithUUID.__init__(self)
+
+        if not all([db_path, table_name]):
+            raise ValueError("cannot specify without db_path and table_name args")
+
+        self.table_name = table_name
+        self.db_path = db_path
+
+        self._key_field = key_field
+        if isinstance(time_field, str) and "." in time_field:
+            self._time_field = time_field.split(".")
```

Comment:
I'm pretty sure we don't support multiple time keys.

Reply:
@gtopper if you sure I will change it